pkg/agents/team2/config.go

```diff
@@ -0,0 +1,126 @@
+		...
+		26				0-30		>60					0-30
+    Action:
+		We separate the TakeFood action into several actions
+
+        Action					Comment
+        disregard food 			takeFood(0), sparing food for agents below me
+		satisfice with food 	takeFood(min), eat only food required for survival
+		satisfy with food   	takeFood(max), eat as manu as possible (get max hp increment)
+	Policy:
+		We associate a probability, or a policy, to a certain action in a certain state.
+    Reward:
+        For MVP, the reward is only related with to hp increment of the agent itself.
+	Q-table:
+		We associate a Q-value to each action under a particular state.
+    Starting State:
+        Initialise policies to a uniform distribution
+		Initialise Q-table to 0
+
+    Episode Termination:
+        Agent gets killed
+		or
+		Simulation terminated
+*/
+
+type actionSpace struct {
+	//TODO: actionId is redundant and may be removed in further version
+	actionId  []int
+	actionSet map[int]func(hp int) food.FoodType
+}
+type CustomAgent2 struct {
+	*infra.Base
+	stateSpace    [][][]int
+	actionSpace   actionSpace
+	policies      [][]float32
+	rTable        [][]float32
+	qTable        [][]float32
+	isPlatArrived bool
+}
+
+func InitTable(numStates int, numActions int) [][]float32 {
+	var table = make([][]float32, numStates)
+	for i := 0; i < numStates; i++ {
+		table[i] = make([]float32, numActions)
+	}
+	return table
+}
+
+func New(baseAgent *infra.Base) (infra.Agent, error) {
+	stateSpace := InitStateSpace(3, 3, 3)
+	actionSpace := InitActionSpace()
+	policies := InitPolicies(27, 3)
+	rTable := InitTable(27, 3)
+	qTable := InitTable(27, 3)
+	return &CustomAgent2{
+		Base:          baseAgent,
+		stateSpace:    stateSpace,
+		actionSpace:   actionSpace,
+		policies:      policies,
+		rTable:        rTable,
+		qTable:        qTable,
+		isPlatArrived: false,
+	}, nil
+}
+
+func (a *CustomAgent2) Run() {
+	//Communication & Observation
+	//communicate before platform arrives to my floor
+
+	/*
+		msg := *messages.NewBaseMessage(a.Floor())
+		a.SendMessage(1, msg)
+	*/
+
+	//Perform the following only when platform arrives
+	//NOTE: should let infra team add a func to see whether the plaftfrom has arrived or not
+	if a.CheckState() != -1 {
+		a.isPlatArrived = true
+	} else {
+		a.isPlatArrived = false
+	}
+	if a.isPlatArrived {
+		oldState := a.CheckState()
+		oldHP := a.HP()
+		a.Log("Agent team2 before action:", infra.Fields{"floor": a.Floor(), "hp": oldHP, "food": a.CurrPlatFood(), "state": oldState})
+		action := a.SelectAction()
+		a.TakeFood(a.actionSpace.actionSet[action](oldHP)) //perform selected action
+		a.Log("Agent team2:", infra.Fields{"selected and performed action": action})
+		a.Log("Agent team2 after action:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "food": a.CurrPlatFood(), "state": a.CheckState()})
+		hpInc := a.HP() - oldHP
+		a.updateRTable(hpInc, oldState, action)
+		a.updateQTable(oldState, action)
+		a.updataPolicies(oldState)
```

Comment:
```suggestion
		a.updatePolicies(oldState)
```
(if you fix other typo instance)